api-service/api-server/src/cs_db.ts

```diff
@@ -221,11 +221,11 @@ export class CrowdsourceDB {
   }
 
   public async getClientJobNextOpenQuestions(
-      client_job_key: string, limit: number): Promise<QuestionToAnswer[]> {
+      client_job_key: string, userNonce: string, limit: number): Promise<QuestionToAnswer[]> {
     db_types.assertClientJobKey(client_job_key)
     const query: spanner.Query = {
       sql: `SELECT q.question_id, q.question, c.answers_per_question,
-              COUNT(a.question_id) as answer_count
+              COUNT(a.question_id) as answer_count, STRING_AGG(a.worker_nonce) as worker_ids
```

Comment:
OK, had some time to think about this a little; and I think the structure of this is great. But instead of `STRING_AGG` which will have problems is the data is too big, and also might have clashes from string concat (depending on the nonce strategy used), I think it would be better to use `COUNTIF(REGEXP_CONTAINS(a.worker_nonce, r"${workerIdsRegexp}") as answered_count`; and in the having clause, then specify  lets us generalize the get next open questions so that clients can do a AND answered_count = 0.

Reply:
Done.

Reply:
Your solution is actually better, I realized there was an edge case mine missed (though I'm not exactly sure why...when the string_agg result is empty, the having clause would never register as true, so there were a few comments with no answers that weren't showing up in the results. But that gets fixed with your change!)